templates/title/show.json.twig

```diff
@@ -0,0 +1,46 @@
+{
+    "@context": "http://schema.org/",
+    "@type": "Book",
+    "@id": "{{ url('title_show', {'id':title.id}) }}",
+    "name": "{{title.title}}"
+    {% if title.pubdate %}, "datePublished": "{{title.pubdate}}" {% endif %},
+    "locationCreated": {
+        "@type": "Place"
+        {% if title.locationOfPrinting %}, "name": "{{title.locationOfPrinting}}" {% endif %},
+        "geo": {
+            "@type": "GeoCoordinates",
+            "latitude": "{{title.locationOfPrinting.latitude}}",
+            "longitude": "{{title.locationOfPrinting.longitude}}"
+        }
+    }
+    {% if title.locationOfPrinting %}, "publisherImprint": "{{title.imprint}}" {% endif %},
+    "author": [
+            {% for titleRole in title.getTitleRoles("Author") %}{% set author = titleRole.person %}{
+            "@type": "Person"
+            {% if (author.firstName) and (author.lastName) %}, "name": "{{ author.firstName }} {{ author.lastName }}"  {% endif %},
```

Comment:
There are 39 cases (at my count) where either first name or last name is null, but no instances where both are null, so rather than require both, you could join them together by putting them in an array and using the `join` function (https://twig.symfony.com/doc/3.x/filters/join.html):

```twig
{{ [author.firstName, author.lastName]|join(' ') }}
```